File: BubbleSort/bubble.cpp
#include <iostream>

#define cin std::cin
#define cout std::cout
#define endl std::endl

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

void bubbleSort(int arr[], int n) {
    for(int i = 0; i < n - 1; i++) {
        for(int j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                swap(arr[j], arr[j+1]);
            }
        }
    }
}

int main() {
    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    int arr[n];

    for(int i = 0; i < n; i++) {
        cout << "Enter an element: ";
        cin >> arr[i];
    }

    bubbleSort(arr, n);

    cout << "Sorted array: " << endl;

    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
}

File: BubbleSort/bubble.lua
function bubbleSort(arr)
    local n = #arr
    for i = 1, n - 1 do
        for j = 1, n - i do
            if arr[j] > arr[j+1] then
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
        end
    end

end

function main()
    io.write("Enter number of elements: ")
    local n = io.read("*n")

    local arr = {}

    for i = 1, n do
        io.write("Enter an element: ")
        arr[i] = io.read("*n")
    end

    bubbleSort(arr)

    print("Sorted array: ")

    for _, i in ipairs(arr) do
        io.write(i.." ")
    end
end

main()
